### ISSUE TITLE
Replace System.Data.SqlClient with Microsoft.Data.SqlClient

### ISSUE THREAD

**Issue by tsahi** [EnterpriseLibrary/data-access-application-block#38]
**Is your feature request related to a problem? Please describe.**
Microsoft [has announced](https://devblogs.microsoft.com/dotnet/introducing-the-new-microsoftdatasqlclient/) that future development of their .NET SQL Server driver will be done in [Microsoft.Data.SqlClient](https://www.nuget.org/packages/Microsoft.Data.SqlClient), released as a NuGet package for both .NET Core and .NET Framework. To keep with the latest developments, we need to use that instead of `System.Data.SqlClient`.

**Describe the solution you'd like**
This is mostly a find-and-replace of the namespace once the package is added.

Microsoft also maintains a NuGet package named [System.Data.SqlClient](https://www.nuget.org/packages/System.Data.SqlClient/), but all their publications promote Microsoft.Data.SqlClient.

**Comment by tsahi** [comment]
I'm contemplating if to replace System.Data.SqlClient with Microsoft.Data.SqlClient, or support both namespaces (as two separate packages), because users might not want to jump over to the Microsfot.Data.SqlClient wagon yet. What do you think?
@Chavoshi

**Comment by Chavoshi** [comment]
Hi @tsahi,
I somehow do not agree with supporting both namespaces in do different packages, but the thing we can do is make it conditional for different frameworks, older frameworks can use the old namespace but .Net Core and new .Net Frameworks (we can decide about the frameworks) should use the new one, what do you think?

**Comment by tsahi** [comment]
Yes, we can do that. We can support the new namespace for the latest frameworks (.NET Framework 4.8 and .NET 5.0), and add a dependency on it for these versions. As the older frameworks go out of support from Microsoft (some already have), we can drop support for them and eventualy, a few years ahead, drop support for the old namespace entirely. Sounds right?

**Comment by Chavoshi** [comment]
Right

**Comment by tsahi** [comment]
We will implement this when we add support for .NET 5 (#41) and .NET Framework 4.8, in a future release. See [comment above](#issuecomment-735353698).

**Comment by wktylertech** [comment]
Is there any update on Replace System.Data.SqlClient with Microsoft.Data.SqlClient ?
